src/mscorlib/src/System/Threading/Tasks/TPLETWProvider.cs

```diff
@@ -528,6 +534,25 @@ public void NewID(int TaskID)
                 WriteEvent(26, TaskID);
         }
 
+        [NonEvent]
+        public void IncompleteAsyncMethod(IAsyncStateMachineBox stateMachineBox)
+        {
+            System.Diagnostics.Debug.Assert(stateMachineBox != null);
+            if (IsEnabled(EventLevel.Warning, Keywords.AsyncMethod))
+            {
+                IAsyncStateMachine stateMachine = stateMachineBox.GetStateMachineObject();
+                if (stateMachine != null)
+                {
+                    string description = AsyncMethodBuilderCore.GetAsyncStateMachineDescription(stateMachine);
+                    IncompleteAsyncMethod(description);
+                }
+            }
+        }
+
+        [Event(27, Level = EventLevel.Warning, Keywords = Keywords.AsyncMethod)]
+        private void IncompleteAsyncMethod(string stateMachineDescription) =>
+            WriteEvent(27, stateMachineDescription);
```

Comment:
@vancem, when I get this event in PerfView, it seems like PerfView doesn't know about the schema for the event and thus doesn't show me the data (unless I look at the raw data view).  If I move the event to a different provider, though, PerfView understands the schema ok.  Does PerfView have a hardcoded knowledge of this TplEventSource, and if so, is it unable to understand new events that aren't a part of the schema when it was originally told about the event source?

Reply:
@stephentoub, yes PerfView does have hardcoded knowledge of TplEventSource, so I believe it will ignore the manifest sent in the trace.  See https://github.com/Microsoft/perfview/blob/08095c84838dd8c9809a3d33796c6d409f3a36e3/src/TraceEvent/Parsers/TplTraceEventParser.cs for details.  It may be easiest to just add the new event here, but I'm also wondering if we can't just remove this hardcoded knowledge and use the manifest from the trace.  @vancem?

Reply:
@stephentoub  can you provide a trace with the new event?   PerfView does have hard-coded knowledge (it can parse the events much more efficiently if it does (it also has a better programmer experience).   I would have expected however that PerfView would use the EventSource manifest on any event where the hard coded knowledge is not present (that is new events).   I need to figure out why that is not true here.

Reply:
Thanks, @vancem.  Sent you a trace.

Reply:
The trace is interesting in that the manifest that the EventSource created dynamically does NOT have event 27 in it.   Thus it looks like an EventSource issue.  I need to look more at it, but you should not block on this.

Reply:
Ok.  Thanks, @vancem and @brianrob.